File: AstroDroidJs.Core/src/commands/drive-command.ts
import { DriveDirection } from "../enums/drive-direction";
import { ICommand } from "../interfaces/command";

export class DriveCommand implements ICommand
{
  direction: DriveDirection = 0;
  distance: number = 200;

  GetName() : string {
    return "DriveCommand";
  }
}

File: AstroDroid.Core/Commands/DriveCommand.cs
using AstroDroid.Core.Interfaces;

namespace AstroDroid.Core.Commands
{
    /// <summary>
    /// Users leverage this command to drive in various directions
    /// </summary>
    public class DriveCommand : INodeCommand
    {
        public DriveDirection Direction { get; set; } = DriveDirection.Forward;
        public float DistanceInMeters { get; set; } = 1;
        public string Name => "DriveCommand";
    }
}